### PR TITLE
Improve loader and engine testing

### PR DESCRIPTION
**Pull request by mwestphal**
- Improve loader and engine testing
- Improve loader and engine doc
- Fix a small issue in loader file list

### CODE REVIEW

File: library/public/engine.h [hunk 1]
```diff
@@ -2,7 +2,18 @@
...
  * @brief   Class used to create instance of other classes
  *
- * TODO improve doc
+ * The main class of the libf3d to create all other instances
+ * Configured on creation using binary flags, then all objects
+ * can be accessed through their getter.
+ *
+ * Example usage:
+ *
+ *  f3d::engine eng(f3d::engine::CREATE_WINDOW | f3d::engine::CREATE_INTERACTOR);
```
Comment (Meakk) [inline library/public/engine.h:11]: I think there are special tags for code in doxygen

File: library/src/window_impl_noRender.cxx [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 #include <vtkNew.h>
 #include <vtkRenderWindow.h>
+#include <vtkVersion.h>
```
Comment (Meakk) [inline library/src/window_impl_noRender.cxx]: is it needed?

File: library/src/window_impl_noRender.cxx [hunk 3]
```diff
@@ -16,6 +17,7 @@ window_impl_noRender::window_impl_noRender(const options& options)
   : window_impl(options)
   , Internals(new window_impl_noRender::internals)
 {
+  this->Internals->RenWin->SetOffScreenRendering(true);
```
Comment (Meakk) [inline library/src/window_impl_noRender.cxx:19]: I don't like the fact we have an OpenGL render window instance if we have a noRender f3d window. Maybe we should implement a `vtkNoRenderWindow` inheriting from `vtkRenderWindow` and doing nothing.

File: library/testing/TestSDKEngineExceptions.cxx [hunk 4]
```diff
@@ -24,9 +28,14 @@ int TestSDKEngineExceptions(int argc, char* argv[])
...
   catch (const f3d::engine::exception& ex)
   {
-    std::cerr << ex.what() << std::endl;
-    exit(EXIT_SUCCESS);
+    std::cout << ex.what() << std::endl;
+    nbCatch++;
   }
 
-  return EXIT_FAILURE;
+  if (nbCatch != 2)
```
Comment (Meakk) [inline library/testing/TestSDKEngineExceptions.cxx:35]: If you look at my last changes for f3d::image test, I did something better. Just display a message and return in the `try` block.

**Review by Meakk — CHANGES_REQUESTED** [review]

File: library/src/window_impl_noRender.cxx (hunk 3 above)
Comment (mwestphal) [reply library/src/window_impl_noRender.cxx:19]: Out of scope of this PR, but that would be better indeed if possible.

File: library/public/engine.h (hunk 1 above)
Comment (mwestphal) [reply library/public/engine.h:11]: I wil add them buyt am still unsure of the way we will handle API documentation. I hate doxygen with a passion.

File: library/testing/TestSDKEngineExceptions.cxx [hunk 5]
```diff
@@ -9,24 +9,31 @@ int TestSDKEngineExceptions(int argc, char* argv[])
 {
   f3d::engine eng(f3d::engine::FLAGS_NONE);
 
+  // test exceptions
+  int nbCatch = 0;
```
Comment (Meakk) [inline library/testing/TestSDKEngineExceptions.cxx:13]: nbCatch is unused

**Pull request merged**